js/components/MilkdropWindow/index.js

```diff
@@ -0,0 +1,61 @@
+import React from "react";
+import butterchurn from "butterchurn";
+import butterchurnPresets from "butterchurn-presets";
+import GenWindow from "../GenWindow";
+
+class MilkdropWindow extends React.Component {
+  constructor() {
+    super();
+  }
+  componentDidMount() {
+    const analyserNode = this.props.mediaPlayer._analyser;
+    const gainNode = this.props.mediaPlayer._gainNode;
+    this.visualizer = butterchurn.createVisualizer(
+      analyserNode.context,
+      this._canvasNode,
+      {
+        width: this._canvasNode.width,
+        height: this._canvasNode.height
+      }
+    );
+    this.visualizer.connectAudio(gainNode);
+    const presets = butterchurnPresets.getPresets();
+    const preset = presets['Geiss - Reaction Diffusion 2'];
+    this.visualizer.loadPreset(preset);
+    this.renderViz();
+  }
+  componentWillUnmount() {
+    this.pauseViz();
+  }
+  renderViz() {
+    this.animationFrameRequest = requestAnimationFrame(() => this.renderViz());
+    this.visualizer.render();
+  }
+  pauseViz () {
+    if (this.animationFrameRequest) {
+      window.cancelAnimationFrame(this.animationFrameRequest);
+      this.animationFrameRequest = null;
+    }
+  }
+  render() {
+    return (
+      <GenWindow title="Milkdrop" close={() => {}} windowId="MILKDROP_WINDOW">
+        <canvas
+          className="draggable"
+          ref={node => (this._canvasNode = node)}
```

Comment:
For high density displays (retina) we might want to consider setting `width` and `height` attributes that are 2x. It might be too computationally expensive, I don't know. This might become relevant: https://github.com/captbaritone/webamp/issues/552

Reply:
Should be possible